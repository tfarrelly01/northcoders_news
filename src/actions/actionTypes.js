export const FETCH_TOPICS_REQUEST = 'FETCH_TOPICS_REQUEST';
export const FETCH_TOPICS_SUCCESS = 'FETCH_TOPICS_SUCCESS';
export const FETCH_TOPICS_ERROR = 'FETCH_TOPICS_ERROR';

export const FETCH_ARTICLES_REQUEST = 'FETCH_ARTICLES_REQUEST';
export const FETCH_ARTICLES_SUCCESS = 'FETCH_ARTICLES_SUCCESS';
export const FETCH_ARTICLES_ERROR = 'FETCH_ARTICLES_ERROR';

export const FETCH_USERS_REQUEST = 'FETCH_USERS_REQUEST';
export const FETCH_USERS_SUCCESS = 'FETCH_USERS_SUCCESS';
export const FETCH_USERS_ERROR = 'FETCH_USERS_ERROR';

export const FETCH_USER_REQUEST = 'FETCH_USER_REQUEST';
export const FETCH_USER_SUCCESS = 'FETCH_USER_SUCCESS';
export const FETCH_USER_ERROR = 'FETCH_USER_ERROR';

export const FETCH_TOPIC_ARTICLES_REQUEST = 'FETCH_TOPIC_ARTICLES_REQUEST';
export const FETCH_TOPIC_ARTICLES_SUCCESS = 'FETCH_TOPIC_ARTICLES_SUCCESS';
export const FETCH_TOPIC_ARTICLES_ERROR = 'FETCH_TOPIC_ARTICLES_ERROR';

export const FETCH_ARTICLE_REQUEST = 'FETCH_ARTICLE_REQUEST';
export const FETCH_ARTICLE_SUCCESS = 'FETCH_ARTICLE_SUCCESS';
export const FETCH_ARTICLE_ERROR = 'FETCH_ARTICLE_ERROR';

export const FETCH_ARTICLE_COMMENTS_REQUEST = 'FETCH_ARTICLE_COMMENTS_REQUEST';
export const FETCH_ARTICLE_COMMENTS_SUCCESS = 'FETCH_ARTICLE_COMMENTS_SUCCESS';
export const FETCH_ARTICLE_COMMENTS_ERROR = 'FETCH_ARTICLE_COMMENTS_ERROR';

export const ARTICLE_VOTE_REQUEST = 'ARTICLE_VOTE_REQUEST';
export const ARTICLE_VOTE_SUCCESS = 'ARTICLE_VOTE_SUCCESS';
export const ARTICLE_VOTE_ERROR = 'ARTICLE_VOTE_ERROR';

export const COMMENT_VOTE_REQUEST = 'COMMENT_VOTE_REQUEST';
export const COMMENT_VOTE_SUCCESS = 'COMMENT_VOTE_SUCCESS';
export const COMMENT_VOTE_ERROR = 'COMMENT_VOTE_ERROR';

export const ADD_COMMENT_REQUEST = 'ADD_COMMENT_REQUEST';
export const ADD_COMMENT_SUCCESS = 'ADD_COMMENT_SUCCESS';
export const ADD_COMMENT_ERROR = 'ADD_COMMENT_ERROR';

export const DELETE_COMMENT_REQUEST = 'DELETE_COMMENT_REQUEST';
export const DELETE_COMMENT_SUCCESS = 'DELETE_COMMENT_SUCCESS';
export const DELETE_COMMENT_ERROR = 'DELETE_COMMENT_ERROR'

