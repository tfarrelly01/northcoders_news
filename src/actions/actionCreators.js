import * as types from './actionTypes';

export const fetchTopicsRequest = () => {
  return {
    type: types.FETCH_TOPICS_REQUEST
  };
};

export const fetchTopicsSuccess = (topics) => {
  return {
    type: types.FETCH_TOPICS_SUCCESS,
    payload: topics
  };
};

export const fetchTopicsError = (error) => {
  return {
    type: types.FETCH_TOPICS_ERROR,
    payload: error
  };
};

export const fetchArticlesRequest = () => {
  return {
    type: types.FETCH_ARTICLES_REQUEST
  };
};

export const fetchArticlesSuccess = (articles) => {
  return {
    type: types.FETCH_ARTICLES_SUCCESS,
    payload: articles
  };
};

export const fetchArticlesError = (error) => {
  return {
    type: types.FETCH_ARTICLES_ERROR,
    payload: error
  };
};

export const fetchUsersRequest = () => {
  return {
    type: types.FETCH_USERS_REQUEST
  };
};

export const fetchUsersSuccess = (users) => {
  return {
    type: types.FETCH_USERS_SUCCESS,
    payload: users
  };
};

export const fetchUsersError = (error) => {
  return {
    type: types.FETCH_USERS_ERROR,
    payload: error
  };
};

export const fetchUserRequest = () => {
  return {
    type: types.FETCH_USER_REQUEST
  };
};

export const fetchUserSuccess = (user) => {
  return {
    type: types.FETCH_USER_SUCCESS,
    payload: user
  };
};

export const fetchUserError = (error) => {
  return {
    type: types.FETCH_USER_ERROR,
    payload: error
  };
};

export const fetchTopicArticlesRequest = () => {
  return {
    type: types.FETCH_TOPIC_ARTICLES_REQUEST
  };
};

export const fetchTopicArticlesSuccess = (articles) => {
  return {
    type: types.FETCH_TOPIC_ARTICLES_SUCCESS,
    payload: articles
  };
};

export const fetchTopicArticlesError = (error) => {
  return {
    type: types.FETCH_TOPIC_ARTICLES_ERROR,
    payload: error
  };
};

export const fetchArticleRequest = () => {
  return {
    type: types.FETCH_ARTICLE_REQUEST
  };
};

export const fetchArticleSuccess = (article) => {
  return {
    type: types.FETCH_ARTICLE_SUCCESS,
    payload: article
  };
};

export const fetchArticleError = (error) => {
  return {
    type: types.FETCH_ARTICLE_ERROR,
    payload: error
  };
};

export const fetchArticleCommentsRequest = () => {
  return {
    type: types.FETCH_ARTICLE_COMMENTS_REQUEST
  };
};

export const fetchArticleCommentsSuccess = (comments) => {
  return {
    type: types.FETCH_ARTICLE_COMMENTS_SUCCESS,
    payload: comments
  };
};

export const fetchArticleCommentsError = (error) => {
  return {
    type: types.FETCH_ARTICLE_COMMENTS_ERROR,
    payload: error
  };
};

export function articleVoteRequest() {
  return {
    type: types.ARTICLE_VOTE_REQUEST,
  };
}

export function articleVoteSuccess(article) {
  return {
    type: types.ARTICLE_VOTE_SUCCESS,
    payload: article
  };
}

export function articleVoteError(error) {
  return {
    type: types.ARTICLE_VOTE_ERROR,
    payload: error
  };
}

export function commentVoteRequest() {
  return {
    type: types.COMMENT_VOTE_REQUEST,
  };
}

export function commentVoteSuccess(comment) {
  return {
    type: types.COMMENT_VOTE_SUCCESS,
    payload: comment
  };
}

export function commentVoteError(error) {
  return {
    type: types.COMMENT_VOTE_ERROR,
    payload: error
  };
}

export const addCommentRequest = () => {
  return {
    type: types.ADD_COMMENT_REQUEST
  };
};

export const addCommentSuccess = (comment) => {
  return {
    type: types.ADD_COMMENT_SUCCESS,
    payload: comment
  };
};

export const addCommentError = (error) => {
  return {
    type: types.ADD_COMMENT_ERROR,
    payload: error
  };
};

export const deleteCommentRequest = () => {
  return {
    type: types.DELETE_COMMENT_REQUEST
  };
};

export const deleteCommentSuccess = (commentId, message) => {
  return {
    type: types.DELETE_COMMENT_SUCCESS,
    commentId,
    payload: message
  };
};

export const deleteCommentError = (error) => {
  return {
    type: types.DELETE_COMMENT_ERROR,
    payload: error
  };
};
