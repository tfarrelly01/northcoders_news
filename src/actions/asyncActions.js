import axios from 'axios';
import { ROOT } from '../config';
import * as actions from './actionCreators';

// GET api/topics
export const fetchTopics = () => {
  return dispatch => {
    dispatch(actions.fetchTopicsRequest());
    axios.get(`${ROOT}/topics`)
      .then(res => {
        dispatch(actions.fetchTopicsSuccess(res.data.topics));
      })
      .catch(err => {
        dispatch(actions.fetchTopicsError(err));
      });
  };
};

// GET api/articles
export const fetchArticles = () => {
  return dispatch => {
    dispatch(actions.fetchArticlesRequest());
    axios.get(`${ROOT}/articles`)
      .then(res => {
        dispatch(actions.fetchArticlesSuccess(res.data.articles));
      })
      .catch(err => {
        dispatch(actions.fetchArticlesError(err));
      });
  };
};

// GET api/users
export const fetchUsers = () => {
  return dispatch => {
    dispatch(actions.fetchUsersRequest());
    axios.get(`${ROOT}/users`)
      .then(res => {
        dispatch(actions.fetchUsersSuccess(res.data.users));
      })
      .catch(err => {
        dispatch(actions.fetchUsersError(err));
      });
  };
};

// GET api/users/:username
export const fetchUser = (username) => {
  return dispatch => {
    dispatch(actions.fetchUserRequest());
    axios.get(`${ROOT}/users/${username}`)
      .then(res => {
        dispatch(actions.fetchUserSuccess(res.data.user));
      })
      .catch(err => {
        dispatch(actions.fetchUserError(err));
      });
  };
};

// GET api/topics/:topic_slug/articles
export const fetchTopicArticles = (topicSlug) => {
  return dispatch => {
    dispatch(actions.fetchTopicArticlesRequest());
    axios.get(`${ROOT}/topics/${topicSlug}/articles`)
      .then(res => {
        dispatch(actions.fetchTopicArticlesSuccess(res.data.articles));
      })
      .catch(err => {
        dispatch(actions.fetchTopicArticlesError(err));
      });
  };
};

// GET api/articles/:article_id
export const fetchArticle = (articleId) => {
  return dispatch => {
    dispatch(actions.fetchArticleRequest());
    axios.get(`${ROOT}/articles/${articleId}`)
      .then(res => {
        dispatch(actions.fetchArticleSuccess(res.data.article));
      })
      .catch(err => {
        dispatch(actions.fetchArticleError(err));
      });
  };
};

// GET api/articles/:article_id then GET api/users/:username
export const fetchArticleAndUser = (articleId) => {
  return dispatch => {
      let article;
      dispatch(actions.fetchArticleRequest());
      axios.get(`${ROOT}/articles/${articleId}`)
        .then(res => {
          article = res.data.article;
          dispatch(actions.fetchUserRequest());
          let user = axios.get(`${ROOT}/users/${article.created_by}`)
            .then(res => {
              return res.data.user;
            })
            .catch(err => {
              dispatch(actions.fetchUserError(err));
            })
        
          Promise.all([article, user])
          .then(([article, user]) => {       
            dispatch(actions.fetchArticleSuccess(article));
            dispatch(actions.fetchUserSuccess(user));
          })
          .catch(err => {
            dispatch(actions.fetchArticleError(err));
          })  
      })
  };
}

// GET api/articles/:article_id/comments
export const fetchArticleComments = (articleId) => {
  return dispatch => {
    dispatch(actions.fetchArticleCommentsRequest());
    axios.get(`${ROOT}/articles/${articleId}/comments`)
      .then(res => {
        dispatch(actions.fetchArticleCommentsSuccess(res.data.comments));
      })
      .catch(err => {
        dispatch(actions.fetchArticleCommentsError(err));
      });
  };
};

// PUT api/articles/:article_id
export const articleVote = (articleId, vote) => {
  return dispatch => {
    dispatch(actions.articleVoteRequest());
    axios.put(`${ROOT}/articles/${articleId}/?vote=${vote}`)
      .then(res => {
        dispatch(actions.articleVoteSuccess(res.data.article));
      })
      .catch(err => {
        dispatch(actions.articleVoteError(err));
      });
  };
};

// PUT api/comments/:comment_id
export const commentVote = (comment_id, vote) => {
  return dispatch => {
    dispatch(actions.commentVoteRequest());
    axios.put(`${ROOT}/comments/${comment_id}/?vote=${vote}`)
      .then(res => {
        dispatch(actions.commentVoteSuccess(res.data.comment));
      })
      .catch(err => {
        dispatch(actions.commentVoteError(err));
      });
  };
};

// POST api/articles/:article_id/comments
export const addComment = (articleId, comment) => {
  return dispatch => {
    dispatch(actions.addCommentRequest());
    axios.post(`${ROOT}/articles/${articleId}/comments`, { body: comment })
      .then(res => {
        dispatch(actions.addCommentSuccess(res.data.comment));
      })
      .catch(err => {
        dispatch(actions.addCommentError(err));
      });
  };
};

// DELETE api/comments/:comment_id
export const deleteComment = (commentId) => {
  return dispatch => {
    dispatch(actions.deleteCommentRequest());
    axios.delete(`${ROOT}/comments/${commentId}`)
      .then(res => {
        dispatch(actions.deleteCommentSuccess(commentId, res.data));
      })
      .catch(err => {
        dispatch(actions.deleteCommentError(err));
      });
  };
};


