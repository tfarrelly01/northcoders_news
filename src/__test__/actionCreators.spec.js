import * as actions from '../actions/actionCreators';
import * as asyncActions from '../actions/asyncActions';

const data = [
  'foo',
  'bar',
  'baz'
];
const comment = 'I am a comment';
const errMsg = 'ERROR - Error message';

describe('Test Action Creators ', () => {

	describe('Action Creator: fetchTopics', () => {
		test('is a function', () => {
			expect(typeof asyncActions.fetchTopics).toBe('function');
		});

		test('request action returns an object', () => {
			expect(typeof actions.fetchTopicsRequest()).toEqual('object');
		});

		test('success action returns an object', () => {
			expect(typeof actions.fetchTopicsSuccess()).toEqual('object');
		});

		test('error action returns an object', () => {
			expect(typeof actions.fetchTopicsError()).toEqual('object');
		});

		test('success action returns data passed', () => {
			const test = actions.fetchTopicsSuccess(data);
      expect(test).toMatchSnapshot();
		});

		test('error action returns data passed', () => {
			const test = actions.fetchTopicsError(errMsg);
      expect(test.payload).toEqual(errMsg);
		});
	});

	describe('Action Creator: fetchArticles', () => {
		test('is a function', () => {
			expect(typeof asyncActions.fetchArticles).toBe('function');
		});

		test('request action returns an object', () => {
			expect(typeof actions.fetchArticlesRequest()).toEqual('object');
		});

		test('success action returns an object', () => {
			expect(typeof actions.fetchArticlesSuccess()).toEqual('object');
		});

		test('error action returns an object', () => {
			expect(typeof actions.fetchArticlesError()).toEqual('object');
		});

		test('success action returns data passed', () => {
			const test = actions.fetchArticlesSuccess(data);
      expect(test).toMatchSnapshot();
		});

		test('error action returns data passed', () => {
			const test = actions.fetchArticlesError(errMsg);
      expect(test.payload).toEqual(errMsg);
		});
	});

	describe('Action Creator: fetchUsers', () => {
		test('is a function', () => {
			expect(typeof asyncActions.fetchUsers).toBe('function');
		});

		test('request action returns an object', () => {
			expect(typeof actions.fetchUsersRequest()).toEqual('object');
		});

		test('success action returns an object', () => {
			expect(typeof actions.fetchUsersSuccess()).toEqual('object');
		});

		test('error action returns an object', () => {
			expect(typeof actions.fetchUsersError()).toEqual('object');
		});

		test('success action returns data passed', () => {
			const test = actions.fetchUsersSuccess(data);
      expect(test).toMatchSnapshot();
		});

		test('error action returns data passed', () => {
			const test = actions.fetchUsersError(errMsg);
      expect(test.payload).toEqual(errMsg);
		});
	});	
	
	describe('Action Creator: fetchUser', () => {
		test('is a function', () => {
			expect(typeof asyncActions.fetchUser).toBe('function');
		});

		test('request action returns an object', () => {
			expect(typeof actions.fetchUserRequest()).toEqual('object');
		});

		test('success action returns an object', () => {
			expect(typeof actions.fetchUserSuccess()).toEqual('object');
		});

		test('error action returns an object', () => {
			expect(typeof actions.fetchUserError()).toEqual('object');
		});

		test('success action returns data passed', () => {
			const test = actions.fetchUserSuccess(data);
      expect(test).toMatchSnapshot();
		});

		test('error action returns data passed', () => {
			const test = actions.fetchUserError(errMsg);
      expect(test.payload).toEqual(errMsg);
		});
	});

	describe('Action Creator: fetchTopicArticles', () => {
		test('is a function', () => {
			expect(typeof asyncActions.fetchTopicArticles).toBe('function');
		});

		test('request action returns an object', () => {
			expect(typeof actions.fetchTopicArticlesRequest()).toEqual('object');
		});

		test('success action returns an object', () => {
			expect(typeof actions.fetchTopicArticlesSuccess()).toEqual('object');
		});

		test('error action returns an object', () => {
			expect(typeof actions.fetchTopicArticlesError()).toEqual('object');
		});

		test('success action returns data passed', () => {
			const test = actions.fetchTopicArticlesSuccess(data);
      expect(test).toMatchSnapshot();
		});

		test('error action returns data passed', () => {
			const test = actions.fetchTopicArticlesError(errMsg);
      expect(test.payload).toEqual(errMsg);
		});
	});

	describe('Action Creator: fetchArticle', () => {
		test('is a function', () => {
			expect(typeof asyncActions.fetchArticle).toBe('function');
		});

		test('request action returns an object', () => {
			expect(typeof actions.fetchArticleRequest()).toEqual('object');
		});

		test('success action returns an object', () => {
			expect(typeof actions.fetchArticleSuccess()).toEqual('object');
		});

		test('error action returns an object', () => {
			expect(typeof actions.fetchArticleError()).toEqual('object');
		});

		test('success action returns data passed', () => {
			const test = actions.fetchArticleSuccess(data);
      expect(test).toMatchSnapshot();
		});

		test('error action returns data passed', () => {
			const test = actions.fetchArticleError(errMsg);
      expect(test.payload).toEqual(errMsg);
		});
	});

	describe('Action Creator: fetchArticleComments', () => {
		test('is a function', () => {
			expect(typeof asyncActions.fetchArticleComments).toBe('function');
		});

		test('request action returns an object', () => {
			expect(typeof actions.fetchArticleCommentsRequest()).toEqual('object');
		});

		test('success action returns an object', () => {
			expect(typeof actions.fetchArticleCommentsSuccess()).toEqual('object');
		});

		test('error action returns an object', () => {
			expect(typeof actions.fetchArticleCommentsError()).toEqual('object');
		});

		test('success action returns data passed', () => {
			const test = actions.fetchArticleCommentsSuccess(data);
      expect(test).toMatchSnapshot();
		});

		test('error action returns data passed', () => {
			const test = actions.fetchArticleCommentsError(errMsg);
      expect(test.payload).toEqual(errMsg);
		});
	});

	describe('Action Creator: articleVote', () => {
		test('is a function', () => {
			expect(typeof asyncActions.articleVote).toBe('function');
		});

		test('request action returns an object', () => {
			expect(typeof actions.articleVoteRequest()).toEqual('object');
		});

		test('success action returns an object', () => {
			expect(typeof actions.articleVoteSuccess()).toEqual('object');
		});

		test('error action returns an object', () => {
			expect(typeof actions.articleVoteError()).toEqual('object');
		});

		test('success action returns data passed', () => {
			const test = actions.articleVoteSuccess(data);
      expect(test).toMatchSnapshot();
		});

		test('error action returns data passed', () => {
			const test = actions.articleVoteError(errMsg);
      expect(test.payload).toEqual(errMsg);
		});
	});

	describe('Action Creator: commentVote', () => {
		test('is a function', () => {
			expect(typeof asyncActions.commentVote).toBe('function');
		});

		test('request action returns an object', () => {
			expect(typeof actions.commentVoteRequest()).toEqual('object');
		});

		test('success action returns an object', () => {
			expect(typeof actions.commentVoteSuccess()).toEqual('object');
		});

		test('error action returns an object', () => {
			expect(typeof actions.commentVoteError()).toEqual('object');
		});

		test('success action returns data passed', () => {
			const test = actions.commentVoteSuccess(data);
      expect(test).toMatchSnapshot();
		});

		test('error action returns data passed', () => {
			const test = actions.commentVoteError(errMsg);
      expect(test.payload).toEqual(errMsg);
		});
	});

	describe('Action Creator: addComment', () => {
		test('is a function', () => {
			expect(typeof asyncActions.addComment).toBe('function');
		});

		test('request action returns an object', () => {
			expect(typeof actions.addCommentRequest()).toEqual('object');
		});

		test('success action returns an object', () => {
			expect(typeof actions.addCommentSuccess()).toEqual('object');
		});

		test('error action returns an object', () => {
			expect(typeof actions.addCommentError()).toEqual('object');
		});

		test('success action returns data passed', () => {
			const test = actions.addCommentSuccess(comment);
      expect(test).toMatchSnapshot();
		});

		test('error action returns data passed', () => {
			const test = actions.addCommentError(errMsg);
      expect(test.payload).toEqual(errMsg);
		});
	});

	describe('Action Creator: deleteComment', () => {
		test('is a function', () => {
			expect(typeof asyncActions.deleteComment).toBe('function');
		});

		test('request action returns an object', () => {
			expect(typeof actions.deleteCommentRequest()).toEqual('object');
		});

		test('success action returns an object', () => {
			expect(typeof actions.deleteCommentSuccess()).toEqual('object');
		});

		test('error action returns an object', () => {
			expect(typeof actions.deleteCommentError()).toEqual('object');
		});

		test('success action returns data passed', () => {
			const test = actions.deleteCommentSuccess(comment);
      expect(test).toMatchSnapshot();
		});

		test('error action returns data passed', () => {
			const test = actions.deleteCommentError(errMsg);
      expect(test.payload).toEqual(errMsg);
		});
	});

});
