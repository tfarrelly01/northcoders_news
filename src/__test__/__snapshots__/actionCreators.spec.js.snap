// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Action Creators  Action Creator: addComment success action returns data passed 1`] = `
Object {
  "payload": "I am a comment",
  "type": "ADD_COMMENT_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: articleVote success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "ARTICLE_VOTE_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: commentVote success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "COMMENT_VOTE_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: deleteComment success action returns data passed 1`] = `
Object {
  "commentId": "I am a comment",
  "payload": undefined,
  "type": "DELETE_COMMENT_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: fetchArticle success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "FETCH_ARTICLE_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: fetchArticleComments success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "FETCH_ARTICLE_COMMENTS_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: fetchArticles success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "FETCH_ARTICLES_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: fetchTopicArticles success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "FETCH_TOPIC_ARTICLES_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: fetchTopics success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "FETCH_TOPICS_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: fetchUser success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "FETCH_USER_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: fetchUsers success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "FETCH_USERS_SUCCESS",
}
`;
